src/AttributeReader.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+namespace JMS\Serializer;
+
+use Doctrine\Common\Annotations\Reader;
+use ReflectionClass;
+use ReflectionMethod;
+use ReflectionProperty;
+
+class AttributeReader implements Reader
+{
+    public function getClassAnnotations(ReflectionClass $class)
+    {
+        $result = [];
+        $attributes = $class->getAttributes();
+        foreach ($attributes as $attribute) {
+            $result[] = $attribute->newInstance();
+        }
+
+        return $result;
+    }
+
+    public function getClassAnnotation(ReflectionClass $class, $annotationName)
+    {
+        $attributes = $class->getAttributes($annotationName);
+        if (0 === count($attributes) || null === $attributes[0]) {
```

Comment:
Same as https://github.com/schmittjoh/serializer/pull/1332#discussion_r679125075

Wouldn't it be better without calling `count`?